src/main/java/com/bravo/user/service/UserService.java

```diff
@@ -40,6 +41,24 @@ public UserReadDto create(final UserSaveDto request){
     return resourceMapper.convertUser(user);
   }
 
+  public List<UserReadDto> retrieveByName(
+          final String name,
+          final PageRequest pageRequest,
+          final HttpServletResponse httpResponse){
+    UserFilter filter = new UserFilter();
+    String[] names = name.split(" ");
+
+    filter.setFirstNames(Set.of(names[0]));
+    if(names.length == 2) {
+      filter.setLastNames(Set.of(names[1]));
+    } else if(names.length == 3) {
+      filter.setMiddleNames(Set.of(names[1]));
+      filter.setLastNames(Set.of(names[2]));
+    }
+
+    return retrieve(filter, pageRequest, httpResponse);
```

Comment:
Glad you're using the existing service method. However, the UserSpecification implementation is currently doing AND queries against the specific fields you provide. So this implementation would only work for searching users with the same "first", "first and last", or "first, middle, and last" names. It doesn't support searching users that have shared names in different positions. e.g. user1.first == user2.middle